### PR TITLE
Support explain plan statement.

### PR DESCRIPTION
**Pull request by boyjoy1127**
Ref #27141.

Changes proposed in this pull request:
  - Support explain plan statement.

---

Before committing this PR, I'm sure that I have checked the following options:
- [X] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X] I have self-reviewed the commit code.
- [X] I have (or in comment I request) added corresponding labels for the pull request.
- [X] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [X] I have made corresponding changes to the documentation.
- [X] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by sandynz** [comment]
I'm not sure why so much lines of code were removed in `OracleDMLStatementVisitor`, looks they were move to `OracleStatementVisitor`, it needs another review.

**Comment by wgy8283335** [comment]
> I'm not sure why so much lines of code were removed in `OracleDMLStatementVisitor`, looks they were move to `OracleStatementVisitor`, it needs another review.

Because explain needs to visit 'update', 'delete', 'select', 'insert'. So I move them into the parent class, inorder to visit them.
For this solution, I refer to Mysql's solution. MysqlStatementVisitor.

> I'm not sure why so much lines of code were removed in `OracleDMLStatementVisitor`, looks they were move to `OracleStatementVisitor`, it needs another review.

**Comment by sandynz** [comment]
Hi @wgy8283335 , could you help to review it?

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java [hunk 1]
```diff
@@ -850,4 +943,775 @@ public final ASTNode visitDataTypeLength(final DataTypeLengthContext ctx) {
     protected String getOriginalText(final ParserRuleContext ctx) {
         return ctx.start.getInputStream().getText(new Interval(ctx.start.getStartIndex(), ctx.stop.getStopIndex()));
     }
+    
+    @Override
+    public ASTNode visitSelect(final SelectContext ctx) {
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java:948]: Can you use `new OracleDMLStatementVisitor().visit()`?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java (hunk 1 above)
Comment (strongduanmu) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java:948]: Besides, we should merge `private final Collection<ParameterMarkerSegment> parameterMarkerSegments = new LinkedList<>();` if we use `new OracleDMLStatementVisitor().visit()`.

Comment (wgy8283335) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java]: OK

Comment (boyjoy1127) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java]: The codes have been refactored.

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java [hunk 2]
```diff
@@ -32,4 +35,21 @@ public final class OracleDALStatementVisitor extends OracleStatementVisitor impl
...
+        if (null != ctx.insert()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.insert()));
+        } else if (null != ctx.delete()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.delete()));
+        } else if (null != ctx.update()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.update()));
+        } else if (null != ctx.select()) {
+            result.setStatement((SQLStatement) visitor.visit(ctx.select()));
+        }
+        return result;
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java:53]: May be you should call `result.addParameterMarkerSegments(getParameterMarkerSegments());` before return result.

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java [hunk 3]
```diff
@@ -32,4 +35,21 @@ public final class OracleDALStatementVisitor extends OracleStatementVisitor impl
     public ASTNode visitAlterResourceCost(final AlterResourceCostContext ctx) {
         return new OracleAlterResourceCostStatement();
     }
+    
+    @Override
+    public ASTNode visitExplain(final ExplainContext ctx) {
+        OracleExplainStatement result = new OracleExplainStatement();
+        OracleDMLStatementVisitor visitor = new OracleDMLStatementVisitor();
+        visitor.setParameterMarkerSegments(getParameterMarkerSegments());
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java:43]: Maybe we should merge parameterMarkerSegments in OracleDMLStatementVisitor into parameterMarkerSegments in OracleDALStatementVisitor.

```
getParameterMarkerSegments().addAll(visitor.getParameterMarkerSegments());
```

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java [hunk 4]
```diff
@@ -152,7 +152,11 @@
 @Getter(AccessLevel.PROTECTED)
 public abstract class OracleStatementVisitor extends OracleStatementBaseVisitor<ASTNode> {
     
-    private final Collection<ParameterMarkerSegment> parameterMarkerSegments = new LinkedList<>();
+    private Collection<ParameterMarkerSegment> parameterMarkerSegments = new LinkedList<>();
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java:155]: Please do not remove this final modifier.

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java [hunk 5]
```diff
@@ -152,7 +152,11 @@
...
+    
+    public void setParameterMarkerSegments(final Collection<ParameterMarkerSegment> parameterMarkerSegments) {
```
Comment (strongduanmu) [inline parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/OracleStatementVisitor.java:157]: Maybe we don't need setParameterMarkerSegments method.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java (hunk 3 above)
Comment (boyjoy1127) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java:43]: Why should we merge？

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java (hunk 2 above)
Comment (boyjoy1127) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java:53]: Why?

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java (hunk 3 above)
Comment (strongduanmu) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java:43]: If the parameters are not merged, the number of parameters in the parsing result will be less than the actual value.

File: parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java (hunk 2 above)
Comment (strongduanmu) [reply parser/sql/dialect/oracle/src/main/java/org/apache/shardingsphere/sql/parser/oracle/visitor/statement/type/OracleDALStatementVisitor.java:53]: These parameter segments are the content of this sql, if not set, the parsed sql will lose parameters.

**Comment by boyjoy1127** [comment]
I've fixed the pr. @strongduanmu

**Review by strongduanmu — APPROVED** [review]
Good job, merged.

**Pull request merged**